holidays/long_weekends_finder.py

```diff
@@ -0,0 +1,141 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+import calendar
+from datetime import timedelta, date
+
+import holidays
+
+
+def find_long_weekends(country, year, month=None, language=None, exact_range=False):
+    """
+    Find long weekends based on holidays for a specified country and year.
+
+    Args:
+        country (str): The country code to get holidays for.
+        year (int): The year to calculate long weekends for (between 1900 and 2100).
+        month (int, optional): The specific month to filter long weekends (1-12).
+        language (str, optional): The language for holiday names.
+        exact_range (bool, optional): If True, only includes long weekends entirely
+                                      within the specified month.
+
+    Returns:
+        list: A list of dictionaries, each containing:
+            - start_date (date): The start date of the long weekend
+            - end_date (date): The end date of the long weekend
+            - duration (int): The number of days in the long weekend
+            - holidays (list): The names of holidays included in the long weekend
+
+    Raises:
+        ValueError: If year, month, or country is invalid.
+    """
+
+    # Validate country
+    try:
+        holiday_obj = holidays.country_holidays(country, years=year, language=language)
+    except Exception as e:
+        raise ValueError(f"Error fetching holidays: {e}")
```

Comment:
This check is also unnecessary - we catch an exception from another function of our package to throw another exception.